### ISSUE TITLE
boto.route53.healthcheck.HealthCheck not working with TCP

### ISSUE THREAD

**Issue by Antonio-usa2015** [boto/boto#2877]
Hi,

   I"m trying to create Health Checks and when type is HTTP, it works but if type is TCP it doesn't.
   For instance, this works:
   newhc = boto.route53.healthcheck.HealthCheck(ip_addr="5.6.7.8", port=80, hc_type="HTTP", 
                                                resource_path="/", fqdn=None, string_match=None,  request_interval=30, failure_threshold=3)
   AmazonConnection.create_health_check(newhc)

But, this is not working!
   newhc = boto.route53.healthcheck.HealthCheck(ip_addr="5.6.7.8", port=43, hc_type="TCP", 
                                               fqdn=None, string_match=None,  request_interval=30, failure_threshold=3)
   AmazonConnection.create_health_check(newhc)

   Is there a bug???

**Comment by Captainkrtek** [comment]
Hi, what version of boto are you using?

**Comment by Captainkrtek** [comment]
With my fix I am able to get this working now as expected, this was a valid bug as far as I understand. See fix in #2879 
![screen shot 2015-01-12 at 21 15 38](https://cloud.githubusercontent.com/assets/430428/5716314/b977e50e-9aa0-11e4-9864-8e85303f5602.png)

```
>>> newhc = boto.route53.healthcheck.HealthCheck(ip_addr="5.6.7.8", port=80, hc_type="TCP", fqdn=None, string_match=None, request_interval=30, failure_threshold=3)
>>> conn = boto.route53.connect_to_region('universal')
>>> newhc = boto.route53.healthcheck.HealthCheck(ip_addr="5.6.7.8", port=80, hc_type="HTTP",
... resource_path="/", fqdn=None, string_match=None, request_interval=30, failure_threshold=3)
>>> conn.create_health_check(newhc)                                                                                                                                                                         {u'CreateHealthCheckResponse': {u'HealthCheck': {u'HealthCheckConfig': {u'IPAddress': u'5.6.7.8', u'ResourcePath': u'/', u'RequestInterval': u'30', u'Type': u'HTTP', u'Port': u'80', u'FailureThreshold': u'3'}, u'CallerReference': u'471cdcad-8850-47c5-a5af-544606a09fb9', u'HealthCheckVersion': u'1', u'Id': u'2eae1348-a37e-47c9-ab9c-7c981ff43530'}}}
>>> newhc = boto.route53.healthcheck.HealthCheck(ip_addr="5.6.7.8", port=43, hc_type="TCP",
... fqdn=None, string_match=None, request_interval=30, failure_threshold=3)
>>> conn.create_health_check(newhc)                                                                                                                                                                         {u'CreateHealthCheckResponse': {u'HealthCheck': {u'HealthCheckConfig': {u'Type': u'TCP', u'RequestInterval': u'30', u'IPAddress': u'5.6.7.8', u'Port': u'43', u'FailureThreshold': u'3'}, u'CallerReference': u'7c327c11-800c-4402-94b4-5606b0835032', u'HealthCheckVersion': u'1', u'Id': u'62cff963-81d9-4f37-adb1-5a2d1e07ccea'}}}
```

**Comment by Captainkrtek** [comment]
Noticed there is already a PR #2866 which provides a similar fix

**Comment by Antonio-usa2015** [comment]
With Fix #2879 worked like a charm, thanks!
